std-bits/table/src/main/java/org/enso/table/data/column/builder/InferredBuilder.java

```diff
@@ -160,11 +167,25 @@ private void retypeAndAppend(Object o) {
       }
     }
 
+    if (allowDateToDateTimeConversion
+        && o instanceof LocalDate
+        && currentBuilder.canRetypeTo(DateTimeType.INSTANCE)) {
+      currentBuilder = currentBuilder.retypeTo(DateTimeType.INSTANCE);
+      currentBuilder.append(o);
+      return;
+    }
+
     retypeToMixed();
     currentBuilder.append(o);
   }
 
   private void retypeToMixed() {
+    // If the current builder can be re-typed to AnyObjectType, we do that.
+    if (currentBuilder.canRetypeTo(AnyObjectType.INSTANCE)) {
+      currentBuilder = currentBuilder.retypeTo(AnyObjectType.INSTANCE);
+      return;
+    }
```

Comment:
This seems bad. We now have 2 ways of retyping to mixed - through `retypeTo` and `retypeToMixed`.

I think we should either completely remove `retypeToMixed` and make it part of `retypeTo` OR remove the override of `retypeTo` in `DateTimeBuilder` and move that logic to an override of `retypeToMixed`.